File: src/Health.Core/Entities/Meal.cs
using System;
using System.Collections.Generic;

namespace Health.Core.Entities
{
    public class Meal
    {
        public int Id { get; set; }
        public int MealNumber { get; set; }
        public int DayId { get; set; }
        public virtual ICollection<MealEntry> MealEntries { get; set; }
    }
}

File: src/Health.FrontEnd/src/app/nutrition/shared/dtos/meal.ts
import { MealEntry } from "app/nutrition/shared/dtos/mealEntry";

export class Meal {
    public id = 0;
    public mealNumber = 0;
    public dayId = 0;
    public mealEntries: MealEntry[] = [];

    public constructor(init?: Partial<Meal>) {
        Object.assign(this, init);
    }
}